core/commands/storage/upload/upload/dc_repair_router.go

```diff
@@ -0,0 +1,265 @@
+package upload
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/TRON-US/go-btfs/core/commands/cmdenv"
+	"github.com/TRON-US/go-btfs/core/commands/storage/helper"
+	uh "github.com/TRON-US/go-btfs/core/commands/storage/upload/helper"
+	"github.com/TRON-US/go-btfs/core/corehttp/remote"
+
+	"github.com/tron-us/go-btfs-common/crypto"
+	guardpb "github.com/tron-us/go-btfs-common/protos/guard"
+	"github.com/tron-us/go-btfs-common/utils/grpc"
+
+	cmds "github.com/TRON-US/go-btfs-cmds"
+)
+
+var StorageDcRepairRouterCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Interact with Host repair requests and responses for negotiation of shards repair.",
+		ShortDescription: `
+Guard broadcasts repair request to potential nodes, after negotiation, nodes prepare the contracts for 
+such repair job, sign and send to the guard for confirmation `,
+	},
+	Subcommands: map[string]*cmds.Command{
+		"request":  hostRepairRequestCmd,
+		"response": hostRepairResponseCmd,
+	},
+}
+
+type RepairContractParams struct {
+	FileHash             string
+	FileSize             int64
+	RepairPid            string
+	RepairSignTime       time.Time
+	LostShardHashes      []string
+	RepairContractId     string
+	RepairRewardAmount   int64
+	DownloadContractId   string
+	DownloadRewardAmount int64
+}
+
+var hostRepairRequestCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Negotiate with hosts for repair jobs.",
+		ShortDescription: `
+This command sends request to mining host to negotiate the repair works.`,
+	},
+	Arguments: append([]cmds.Argument{
+		cmds.StringArg("peer-ids", true, false, "Host Peer IDs to send repair requests."),
+	}, hostRepairResponseCmd.Arguments...), // append pass-through arguments
+	RunTimeout: 20 * time.Second,
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		fileHash := req.Arguments[1]
+		repairId := req.Arguments[6]
+		repairContractId := req.Arguments[8]
+		lostShardHashes := strings.Split(req.Arguments[2], ",")
+		fileSize, err := strconv.ParseInt(req.Arguments[3], 10, 64)
+		if err != nil {
+			return err
+		}
+		downloadRewardAmount, err := strconv.ParseInt(req.Arguments[4], 10, 64)
+		if err != nil {
+			return err
+		}
+		repairRewardAmount, err := strconv.ParseInt(req.Arguments[5], 10, 64)
+		if err != nil {
+			return err
+		}
+		err = emptyCheck(fileHash, lostShardHashes, repairId, repairContractId, fileSize, downloadRewardAmount, repairRewardAmount)
+		if err != nil {
+			return nil
+		}
+		cfg, err := cmdenv.GetConfig(env)
+		if err != nil {
+			return err
+		}
+		if !cfg.Experimental.StorageClientEnabled {
```

Comment:
Not sure whether our routing node need to enable these two config, are these two settings used for us to specify our routing node?

Reply:
Adding these on the repairer.